plctestbench/data_manager.py

```diff
@@ -37,6 +43,7 @@ def __init__(self, testbench_settings: dict, user: dict) -> None:
             ReconstructedTrackNode,
             OutputAnalysisNode
         ]
+        self.progress_monitor = testbench_settings["progress_monitor"] if 'progress_monitor' in testbench_settings.keys() else lambda caller: std_tqdm
```

Comment:
Invece di passare il progress monitor nei testbench settings propagandolo poi in tutte le classi, dovremmo usare l'import condizionale di Python. Dovremmo probabilmente fare l'import condizionale in un solo file (per esempio utils) e poi importare negli altri file da lì. L'import condizionale si può fare in tanti modi, anche per dire usando una variabile d'ambiente

Reply:
Ottimo. Implementa pure la modifica e poi io mi adeguo. Tanto l'esigenza mi sembra chiara.

Reply:
Ho visto che tu avevi una catena di lambda complicata in cui prima veniva chiamato il progress_monitor(self) che ritornava una funzione tipo tqdm la quale veniva chiamata a sua volta con (range(bla bla bla)). Non ho capito perché funzionasse così per cui per il momento ho tolto il primo step. Ora viene chiamato direttamente progress_monitor(range(bla bla bla)). Fammi sapere se va bene così o se hai esigenze diverse, nel caso ci sentiamo magari brevemente così mi spieghi e posso aggiustare il codice qui